permission_handler_android/example/lib/main.dart

```diff
@@ -8,6 +9,9 @@ void main() {
       githubURL: 'https://github.com/Baseflow/flutter-permission-handler',
       pubDevURL: 'https://pub.dev/packages/permission_handler',
       pages: [PermissionHandlerWidget.createPage()]));
+
+  // TODO(jweener): what is the proper way to register a platform implementation?
+  PermissionHandlerAndroid.registerWith();
```

Comment:
The proper way to do so is by updating the `pubspec.yaml` of the permission_handler_android package. The `dartPluginClass` field informs the Flutter engine on which class it should call the `registerWith` method.

For example the snippet below informs the Flutter engine to call `registerWith` on the `PermissionHandlerAndroid` Dart class:

```yaml
flutter:
  plugin:
    implements: permission_handler
    platforms:
      android:
        package: com.baseflow.permissionhandler
        pluginClass: PermissionHandlerPlugin
        dartPluginClass: PermissionHandlerAndroid
```